src/common/utils/query/visualization-query.ts

```diff
@@ -29,38 +30,24 @@ import { thread } from "../functional/functional";
 
 const $main = $("main");
 
-function applySeries(essence: Essence, timeShiftEnv: TimeShiftEnv, nestingLevel = 0) {
-  const series = essence.getConcreteSeries();
-  const hasComparison = essence.hasComparison();
+function applySeries(series: List<ConcreteSeries>, timeShiftEnv: TimeShiftEnv, nestingLevel = 0) {
 
   return (query: Expression) => {
     return series.reduce((query, series) => {
-      if (!hasComparison) {
         return query.performAction(
-          series.plywoodExpression(nestingLevel)
+          series.plywoodExpression(nestingLevel, timeShiftEnv)
         );
-      }
-      return query
-        .performAction(series.plywoodExpression(nestingLevel, new CurrentFilter(timeShiftEnv.currentFilter)))
-        .performAction(series.plywoodExpression(nestingLevel, new PreviousFilter(timeShiftEnv.previousFilter)));
     }, query);
   };
 }
 
-function applyDeltaSortExpression(essence: Essence, query: Expression, nestingLevel: number, currentFilter: Expression, { reference, period }: Sort): Expression {
-  if (period !== SeriesDerivation.DELTA) return query;
-  // TODO: FIX for non-measure series
-  return query.apply(getNameWithDerivation(reference, SeriesDerivation.DELTA), $(reference).subtract($(getNameWithDerivation(reference, SeriesDerivation.PREVIOUS))));
-}
-
-function applySort(essence: Essence, sort: Sort, currentFilter: Expression, nestingLevel: number) {
-  return (query: Expression) => {
-    const queryWithReference = applyDeltaSortExpression(essence, query, nestingLevel, currentFilter, sort);
-    return queryWithReference.performAction(new SortExpression({
-      expression: $(sort.reference),
+function applySort(sort: Sort) {
+  return (query: Expression) =>
+    query.performAction(new SortExpression({
+      // TODO: Use plywoodKey from correct series!
+      expression: $(getNameWithDerivation(sort.reference, sort.period)),
```

Comment:
We need some function to searching/creating ConcreteSeries from Sort definition. It will be used only during query generation, so we can use whole Essence and DataCube!